### PR TITLE
Rx State Verification - Bug fixes and redesign

### PR DESCRIPTION
**Pull request by obulkin**
[10.0][FIX] ``medical_prescription_state``: Move field and duplicate record
* Move ``base.kanban.stage`` field ``is_verified`` to child module ``medical_prescription_state_verify``
* Replace second occurrence of the ``prescription_order_state_exception`` record with ``prescription_order_line_state_exception``, an equivalent stage intended for the ``medical.prescription.order.line`` model

[10.0][FIX] ``medical_prescription_state_verify``: Move field
* Move ``base.kanban.stage`` field ``is_verified`` from parent module ``medical_prescription_state``

[10.0][IMP] ``medical_prescription_state_verify``: Redesign
* Update ``base.kanban.stage`` records introduced in ``medical_prescription_state`` to be verified where needed
* Rework verification-related logic in ``medical.prescription.order`` and ``medical.prescription.order.line``
* Add demo data and unit tests for new logic
* Update README and manifest

### CODE REVIEW

**Comment by obulkin** [comment]
This PR is meant to replace #185 and #211.

File: medical_prescription_state_verify/README.rst [hunk 1]
```diff
@@ -3,58 +3,52 @@
...
-Adds a concept of types to MedicalPrescriptionOrderState that may be validated
-on. Defaults with a ``verified`` type that will not allow attribute updates, or
-status changes to types other than ``exception`` and ``cancel``.
+This module introduces the notion of verification to the prescription states 
+added in `medical_prescription_state`, which can now be either verified or 
+unverified. It also adds the following logic:
 
+* Once a prescription order is in a verified state, it can be cancelled or 
+  marked as an exception, but no other changes to the order are allowed
+* When a prescrption order moves from an unverified state to a verified one,
```
Comment (lasley) [inline medical_prescription_state_verify/README.rst]: s/prescrption/prescription

File: medical_prescription_state_verify/models/medical_prescription_order.py [hunk 2]
```diff
@@ -2,61 +2,70 @@
...
+                          ' either cancel them or mark them as exceptions if'
+                          ' manual changes are required.')
+                        % verified_recs[0].name
                     )
-                    if stage_id.name not in self._ALLOWED_CHANGE_STATES:
-                        raise exceptions.ValidationError(_(
-                            'You cannot move an Rx into this state after it'
-                            ' has been verified. [%s]' % rec_id.name
-                        ))
+                else:
```
Comment (lasley) [inline medical_prescription_state_verify/models/medical_prescription_order.py]: You can actually remove this else and eliminate a tad of complexity. The raise is like a return - code execution stops at that point, thus the else is superfluous

**Review by lasley — CHANGES_REQUESTED** [review]
Few minor changes

**Comment by lasley** [comment]
@obulkin - please squash the last three commits into one for merge. I can't do an auto-squash here due to the two modules.

**Comment by obulkin** [comment]
@lasley - The branch is squashed now and ready for a merge.

**Review by lasley — APPROVED** [review]

**Comment by lasley** [comment]
Thanks Oleg!

**Pull request merged**